src/PrestaShopBundle/Install/DatabaseDump.php

```diff
@@ -118,40 +205,189 @@ private function exec($command)
      */
-    public function restore()
+    private function getTableChecksum(string $table): string
     {
-        $restoreCommand = $this->buildMySQLCommand('mysql', [$this->databaseName]);
-        $restoreCommand .= ' < ' . escapeshellarg($this->dumpFile) . ' 2> /dev/null';
-        $this->exec($restoreCommand);
+        $checksum = $this->db->executeS(sprintf('CHECKSUM TABLE %s;', $table));
+        $checksum = $checksum[0]['Checksum'];
+
+        // The content only is not enough we must make sure that the auto increment index is the same
+        $autoIncrement = $this->db->executeS(sprintf(
+            'SELECT AUTO_INCREMENT FROM information_schema.TABLES WHERE TABLE_SCHEMA = "%s" AND TABLE_NAME = "%s";',
+            $this->databaseName,
+            $table
+        ));
+        // Some tables have no auto increment (like relation tables for example)
+        $autoIncrement = (int) ($autoIncrement[0]['AUTO_INCREMENT'] ?? 0);
+
+        return $checksum . $autoIncrement;
+    }
+
+    private function checkDumpFile(): void
+    {
+        if (!file_exists($this->dumpFile)) {
+            throw new Exception('You need to run \'composer create-test-db\' to create the initial test database');
+        }
+    }
+
+    private function checkTableDumpFile(string $tableName): void
+    {
+        $dumpFile = $this->getTableDumpPath($tableName);
+        if (!file_exists($dumpFile)) {
+            throw new Exception(sprintf(
+                'Cannot find dump for table %s, you need to run \'composer create-test-db\' to create the initial test database',
+                $tableName
+            ));
+        }
     }
 
     /**
      * Make a database dump.
      */
-    public static function create()
+    public static function create(): void
     {
         $dump = new static();
 
         $dump->dump();
     }
 
+    /**
+     * Make dump for each table in the database.
+     */
+    public static function dumpTables(): void
+    {
+        $dump = new static();
+
+        $dump->dumpAllTables();
+    }
+
+    /**
+     * Check that dump file exists
+     *
+     * @throws Exception
+     */
+    public static function checkDump(): void
+    {
+        $dump = new static();
+
+        $dump->checkDumpFile();
+    }
+
     /**
      * Restore a database dump.
      */
-    public static function restoreDb()
+    public static function restoreDb(): void
     {
         $dump = new static();
 
         $dump->restore();
     }
+
+    /**
+     * Restore all tables (only modified tables are restored)
+     */
+    public static function restoreAllTables(): void
+    {
+        $dump = new static();
+
+        $tables = $dump->db->executeS('SHOW TABLES;');
+        foreach ($tables as $table) {
+            // $table is an array looking like this [Tables_in_database_name => 'ps_access']
+            $tableName = reset($table);
+            $tableName = substr($tableName, strlen($dump->dbPrefix));
+            $dump->restoreTable($tableName);
```

Comment:
@jolelievre 

one quick question, you're removing the prefix in this part of the code, to add it again in `restoreTable` method, what is the idea behind it? am I missing something?

Reply:
Reading the code further, I understand why. Resolved.